pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -843,16 +813,9 @@ private CompletableFuture<Void> internalRemovePartitionsTopicNoAutocreationDisab
                 }).collect(Collectors.toList()));
     }
 
-    private CompletableFuture<Void> internalRemovePartitionsAuthenticationPoliciesAsync(int numPartitions) {
+    private CompletableFuture<Void> internalRemovePartitionsAuthenticationPoliciesAsync() {
```

Comment:
What is the motivation for removing the partition logic here? I don't think we can do that here, but I might be missing some context.

My understanding is that we need to remove permission for the parent and all of its sub partitions.

Reply:
yes, you're right.  We will `remove permission` in the BrokerService#deleteTopic. So we just need to remove the topic without partitioned here.  
See line 848, I have deleted them together.

Reply:
I saw `removePermissionsAsync` only deleted the parent info but no sub info in `internalRemovePartitionsAuthenticationPoliciesAsync` method. 
Do you mean the rest of the sub info will delete by `BrokerService#deleteTopic`?

Reply:
yes, right, it's duplicated here.